internal/pkg/checkin/bulk.go

```diff
@@ -209,6 +209,11 @@ func (bc *Bulk) flush(ctx context.Context) error {
 				fields[dl.FieldAgent] = map[string]interface{}{
 					dl.FieldAgentVersion: pendingData.extra.ver,
 				}
+				// Duplicates the writes done when an upgrade is ack'd by the agent.
+				// This helps eliminate the case where the agent is stuck in upgrading after it's already upgraded if the ack
+				// is never received.
+				fields[dl.FieldUpgradeStartedAt] = nil
+				fields[dl.FieldUpgradedAt] = nowTimestamp
```

Comment:
Did some local testing, and I think this doesn't work well, the version is coming in every checkin, and the `upgraded_at` field keeps updating, which can be misleading.
Added logging in line 212:

```
07:19:07.751 INF new version ecs.version=1.6.0 service.name=fleet-server ver=8.2.1
07:19:37.751 INF new version ecs.version=1.6.0 service.name=fleet-server ver=8.2.1
07:20:07.750 INF new version ecs.version=1.6.0 service.name=fleet-server ver=8.2.1
07:20:37.749 INF new version ecs.version=1.6.0 service.name=fleet-server ver=8.2.1
07:21:07.748 INF new version ecs.version=1.6.0 service.name=fleet-server ver=8.2.1
```

I think the right thing to do here would be to read the existing action doc, and only update the upgrade fields if the version is newer, and the `upgrade_started_at` field is set.
I'll revert for now.

Reply:
I also think it might mess up the action status tracking in the UI if we don't create the action result document?

Reply:
yes, the action would stay in progress